vulnerability-analyzer/src/main/java/org/hyades/processor/scanner/snyk/SnykProcessor.java

```diff
@@ -171,6 +173,21 @@ public void analyzeBatch(final List<RetryableRecord<String, ScanTask>> batch) {
             cacheReport(mappedReport);
         }
 
+        PageMeta pageMeta = page.meta();
+        if (pageMeta != null && pageMeta.errors() != null) {
+            for (final MetaError metaError : pageMeta.errors()) {
+                if (metaError.meta() == null || metaError.meta().purl() == null) {
+                    continue;
+                }
+                var purlWithError = metaError.meta().purl();
+                final var errorMessage = "Purl " + purlWithError + " failed with status " + metaError.status() +
+                        " due to Snyk error code " + metaError.code() + " : " + metaError.title();
+                LOGGER.error(errorMessage);
+                reportFailure(purlRecords.getFirst(purlWithError), new RuntimeException(errorMessage));
```

Comment:
`purlRecords` is a multi-valued map, which means that one PURL can map to multiple `Record`s. We need to ensure that we send a result for every record, so only handling `purlRecords.getFirst` is insufficient.